### ISSUE TITLE
Error 500 while setting up openvpn

### ISSUE THREAD

**Issue by stueja** [freedombox/Plinth#459]
During the really long generation of the dh4096.pem for openvpn on a BeagleBone Black with `freedombox-unstable-free_2015-12-13_beaglebone-armhf.img`, after ~2 hours, I got the 500 error in the browser. The process itself is still running.
``` bash
ps ax | grep openssl
 4296 ?        R    125:41 openssl dhparam -out /etc/openvpn/freedombox-keys/dh4096.pem 4096
```
`tail -n <xx> /var/log/plinth/status.log`:

``` python
[2016-05-17 21:43:00,547] plinth.actions INFO     Executing command - ['sudo', '-n', '/usr/share/plinth/actions/openvpn', 'is-setup']
[2016-05-17 21:43:00,563] django.request ERROR    Internal Server Error: /plinth/apps/openvpn/
Traceback (most recent call last):
  File "/usr/lib/python3/dist-packages/django/core/handlers/base.py", line 149, in get_response
    response = self.process_exception_by_middleware(e, request)
  File "/usr/lib/python3/dist-packages/django/core/handlers/base.py", line 147, in get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
  File "/usr/lib/python3/dist-packages/plinth/package.py", line 233, in wrapper
    return func(request, *args, **kwargs)
  File "/usr/lib/python3/dist-packages/plinth/modules/openvpn/views.py", line 44, in index
    status = get_status()
  File "/usr/lib/python3/dist-packages/plinth/modules/openvpn/views.py", line 101, in get_status
    status = {'is_setup': openvpn.is_setup(),
  File "/usr/lib/python3/dist-packages/plinth/modules/openvpn/__init__.py", line 59, in is_setup
    return actions.superuser_run('openvpn', ['is-setup']).strip() == 'true'
  File "/usr/lib/python3/dist-packages/plinth/actions.py", line 118, in superuser_run
    return _run(action, options, input, async, True)
  File "/usr/lib/python3/dist-packages/plinth/actions.py", line 172, in _run
    shell=False)
  File "/usr/lib/python3.4/subprocess.py", line 859, in __init__
    restore_signals, start_new_session)
  File "/usr/lib/python3.4/subprocess.py", line 1365, in _execute_child
    errpipe_read, errpipe_write = os.pipe()
OSError: [Errno 24] Too many open files
[2016-05-17 21:53:18,600] plinth.actions INFO     Executing command - ['sudo', '-n', '/usr/share/plinth/actions/openvpn', 'is-setup']
[2016-05-17 21:53:28,456] plinth.actions INFO     Executing command - ['sudo', '-n', '/usr/share/plinth/actions/openvpn', 'is-setup']
[2016-05-17 21:53:28,468] django.request ERROR    Internal Server Error: /plinth/apps/openvpn/
Traceback (most recent call last):
  File "/usr/lib/python3/dist-packages/django/core/handlers/base.py", line 149, in get_response
    response = self.process_exception_by_middleware(e, request)
  File "/usr/lib/python3/dist-packages/django/core/handlers/base.py", line 147, in get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
  File "/usr/lib/python3/dist-packages/plinth/package.py", line 233, in wrapper
    return func(request, *args, **kwargs)
  File "/usr/lib/python3/dist-packages/plinth/modules/openvpn/views.py", line 44, in index
    status = get_status()
  File "/usr/lib/python3/dist-packages/plinth/modules/openvpn/views.py", line 101, in get_status
    status = {'is_setup': openvpn.is_setup(),
  File "/usr/lib/python3/dist-packages/plinth/modules/openvpn/__init__.py", line 59, in is_setup
    return actions.superuser_run('openvpn', ['is-setup']).strip() == 'true'
  File "/usr/lib/python3/dist-packages/plinth/actions.py", line 118, in superuser_run
    return _run(action, options, input, async, True)
  File "/usr/lib/python3/dist-packages/plinth/actions.py", line 172, in _run
    shell=False)
  File "/usr/lib/python3.4/subprocess.py", line 823, in __init__
    errread, errwrite) = self._get_handles(stdin, stdout, stderr)
  File "/usr/lib/python3.4/subprocess.py", line 1313, in _get_handles
    errread, errwrite = os.pipe()
OSError: [Errno 24] Too many open files
```

**Comment by SunilMohanAdapa** [comment]
On 05/18/2016 03:35 AM, Jan wrote:
> During the really long generation of the dh4096.pem for openvpn on a
> BeagleBone Black with
> |freedombox-unstable-free_2015-12-13_beaglebone-armhf.img|, after ~2
> hours, I got the 500 error in the browser. The process itself is still
> running.

There is a known issue with PackageKit Glib that it leaks a file
descriptor every time it is used.  We were using it on every page load
earlier in 0.7.2 images.  Here are a couple of work arounds:

- Upgrade to Plinth 0.9.  It does not use PackageKit that often and
should not face the same issue.

- Start the generation process and navigate away from the OpenVPN page
so that it does not do a lot of page refreshes (leading to more and more
FDs leakings).

There a consideration to not use PackageKit in future that will make the
problem go away.

-- 
Sunil

**Comment by jvalleroy** [comment]
Fixed in #493.
